File: Unidad_2/05_Lambda_ejercicios/02_Lambda_ejercicio_c2/Principal.java
public class Principal{
    public static void main(String[] args) {
        System.out.println("==> CalculadoraInt --> Resultado = " + Principal.engine_1().calculate(5,5));
        System.out.println("==> CalculadoraLong --> Resultado = " + Principal.engine_2().calculate(6,2));
    }
    //Retorna un objeto de tipo "calculardoraInt"
    private static CalculadoraInt engine_1(){
        return (x, y) -> x * y;
    }

    //Retorna un objeto de tipo "calculardoraLong"
    private static CalculadoraLong engine_2(){
        return (x, y) -> x - y;
    }
}

File: 2 _oportunidad/_07_ejercicios/retorno_lambda_v1/Principal.py
# corecto
class Principal:
    @staticmethod
    def main(args):
        #/Principal.engine((int x, int y)->(long x1,long y2)->x1+y2)
        print("CalcularInt --> resultado = "+ str(Principal._engine1().calculate(5,2)))
        print("CalcularLong --> resultado = "+ str(Principal._engine2().calculate(6,2)))
    @staticmethod
    def _engine1():
        return lambda x, y : x * y
    @staticmethod
    def _engine2():
        return lambda x, y : x - y
